WordPressAuthenticator/Email Client Picker/AppSelector.swift

```diff
@@ -0,0 +1,122 @@
+import MessageUI
+import UIKit
+
+/// App selector that selects an app from a list and opens it
+/// Note: it's a wrapper of UIAlertController (which cannot be sublcassed)
+class AppSelector {
+    // the action sheet that will contain the list of apps that can be called
+    let alertController: UIAlertController
+
+    /// initializes the picker with a dictionary. Initialization will fail if an empty/invalid app list is passed
+    /// - Parameters:
+    ///   - appList: collection of apps to be added to the selector
+    ///   - defaultAction: default action, if not nil, will be the first element of the list
+    ///   - urlHandler: object that handles app URL schemes; defaults to UIApplication.shared
+    init?(with appList: [String: String],
+          defaultAction: UIAlertAction? = nil,
+          urlHandler: URLHandler = UIApplication.shared) {
+        /// inline method that builds a list of app calls to be inserted in the action sheet
+        func buildAppCalls(from appList: [String: String]) -> [UIAlertAction]? {
```

Comment:
If I use an instance method, the compiler would complain that I'm using self before initializing. It could be a class method but does not seem that appropriate, since it's only used at initialization time. The other option is to just put that code in the init.
Absolutely agree that it should be make instead of build, and makeAlertActions sounds more appropriate

Reply:
> If I use an instance method, the compiler would complain that I'm using self before initializing

Sorry, my bad – I always get tripped up by this! 🙃